### PR TITLE
Update/version_2a

### PR DESCRIPTION
**Pull request by grayseff**
Significant changes to the structure of the code up to the mc_cycle level. Main changes:
- get_energy function correctly implemented
- Exchange module created to implement metropolis and metropolis-hastings conditions -- declutters MC_Run
- some tidying of redundant versions of functions.
- functionalised and vectorised atom move, get energy and acceptance test functions created
- mc_step now uses these functions
-mc_cycle now uses these and a parallel tempering function. 
-RuNNer version now undergoes correct number of steps

### CODE REVIEW

File: scripts/run_ne13.jl [hunk 1]
```diff
@@ -1,5 +1,5 @@
-using ParallelTemperingMonteCarlo
-using Random,Plots
+using ParallelTemperingMonteCarlo#update/version_2a
+using Random , BenchmarkTools, Plots
```
Comment (ElkePahl) [inline scripts/run_ne13.jl:2]: BenchmarkTools use of it should be removed

File: scripts/test.ipynb [hunk 2]
```diff
@@ -2,133 +2,62 @@
  "cells": [
```
Comment (ElkePahl) [inline scripts/test.ipynb:2]: Is this a Julia notebook? What does it do? I don't think that we want it in source code?

**Review by ElkePahl — COMMENTED** [review]
It might be helpful to get some explanations via Zoom

File: scripts/run_ne13.jl (hunk 1 above)
Comment (grayseff) [reply scripts/run_ne13.jl:2]: Done

**Review by ElkePahl — APPROVED** [review]
Hi Gray,

Happy to approve those changes. In the next update could you please change the one function name "swap_var_function!" to "swap_var!" or better "swap_config!" to be consistent in naming? Thanks

**Pull request merged**